packages/world/src/systems/libraries/MoveLib.sol

```diff
@@ -226,6 +234,19 @@ library MoveLib {
     }
   }
 
+  function _updatePlayerDrainRate(EntityId player, Vec3 finalCoord) private {
+    uint128 drainRate;
+    if (EntityUtils.getObjectTypeAt(finalCoord - vec3(0, 1, 0)) == ObjectTypes.Lava) {
+      drainRate = PLAYER_LAVA_ENERGY_DRAIN_RATE;
+    } else if (_isFluid(finalCoord + vec3(0, 1, 0))) {
```

Comment:
hmm this only checks fluid level right, so I could end my path on a oak log but still get a swim energy rate?

Reply:
no because this is the top coord of the player, there can't be anything else there, only passthrough blocks